### ISSUE TITLE
Anti-Idle: Shift-click does not work

### ISSUE THREAD

**Issue by FelicitusNeko** [ruffle-rs/ruffle#17982]
### Describe the bug

Several elements in Anti-Idle: The Game require to shift-click to perform certain interactions (sell items in Battle Arena, remove trees in garden, confirmation for destructive actions). However, it seems not to detect Shift pressed state.

Steps to reproduce (one example):
- Start a game
- Play long enough to get to the Garden
- Plant a Free tree in the Garden
- Shift-click on the Free tree to remove it
- An error message is displayed that there is already a tree in that plot, and to shift-click to remove it

### Expected behavior

In this case, the tree would be removed from the Garden.

### Content Location

[Anti-Idle v1861.zip](https://github.com/user-attachments/files/17051170/Anti-Idle.v1861.zip)

### Affected platform

Desktop app

### Operating system

Garuda Linux Dr460nized, Wayland/KDE

### Additional information

Tried with latest stable version (2024-08-19) and latest nightly version (2024-09-18) in AUR.

**Comment by kjarosh** [comment]
Hi @FelicitusNeko! I've tried to reproduce this bug, but on the latest nightly (2024-08-19) it seems that everything works, see the recording below. Are you sure you used the desktop app?

[Screencast from 2024-09-19 08-36-26.webm](https://github.com/user-attachments/assets/4d8ef477-0a83-44ea-b595-459db4105c35)

**Comment by FelicitusNeko** [comment]
Yes, I'm using the desktop app. I've tried multiple times to shift-click on things, and it doesn't work for me. Ctrl-click works fine. Some elements such as Battle Arena skill +/- buttons and LolMarket buy/sell have you hold Space and click, and that also works. Only shift-click doesn't.

**Comment by kjarosh** [comment]
Could you run the following SWF and check whether it says `Shift` when shift is pressed? [pressed_keys.zip](https://github.com/user-attachments/files/17059216/pressed_keys.zip)

Also please verify whether you can input uppercase characters while pressing shift in the following SWF: [basic_input.zip](https://github.com/user-attachments/files/17059245/basic_input.zip)

**Comment by FelicitusNeko** [comment]
The basic input one does allow to enter capital letters. Also worth noting that any input in Anti-Idle that requires pressing Shift and a key seems to work.

The pressed keys one shows KeyCode 20 when I press Shift. It does not go away when I release Shift. Pressing Ctrl and Alt show Ctrl and Alt, and those do go away when I release them. Any other key seems to work fine.

**Comment by FelicitusNeko** [comment]
!["Both Shifts together enable Caps Lock; one Shift key disables it" with a checked box](https://github.com/user-attachments/assets/e8cfddf8-fb73-46f9-b2d7-762bed262986)
I had enabled this option in Plasma keyboard settings. Turning it off makes the pressed keys tester properly recognise Shift. Trying it in Anti-Idle, it's now correctly allowing me to sell items or remove trees. It's weird that it would recognise one Shift key as Caps Lock, as the rest of the system seems to properly handle it as Shift being held just fine.
